redskyctl/internal/commands/generate/trial.go

```diff
@@ -121,5 +129,40 @@ func (o *TrialOptions) generate() error {
 	t.Finalizers = nil
 	t.Annotations = nil
 
-	return o.Printer.PrintObj(t, o.Out)
+	// If requested, convert the trial into a job
+	var obj interface{} = t
+	if o.Job != "" {
+		obj, err = newJob(t, o.Job, o.JobTrialNumber)
+		if err != nil {
+			return err
+		}
+	}
+
+	return o.Printer.PrintObj(obj, o.Out)
```

Comment:
I'd prefer to drop the interface to make it a bit more clear. I get that using interface{} helps with only needing to handle a single return, but it made me question it/reread a few times. 

```suggestion
if o.Job == "" {
  return o.Printer.PrintObj(t, o.Out)
}

obj, err := newJob(t, o.Job, o.JobTrialNumber)
if err != nil {
  return err
}
  		
return o.Printer.PrintObj(obj, o.Out)
```